drivers/i2c/i2c_ll_stm32.c

```diff
@@ -286,6 +303,57 @@ static int i2c_stm32_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+
+static int i2c_stm32_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct i2c_stm32_config *config = dev->config;
+	struct i2c_stm32_data *data = dev->data;
+	int err;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		/* Set pins to active state */
+		err = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+		if (err < 0) {
+			return err;
+		}
+		/* enable clock */
+		err = clock_control_on(data->clock, (clock_control_subsys_t)&config->pclken);
+		if (err != 0) {
+			LOG_ERR("Could not enable I2C clock");
+			return err;
+		}
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		while (LL_I2C_IsEnabled(config->i2c) == 1) {
+		}
```

Comment:
This probably works but feels like a hack on `pm_device_action_run`, while PM subsystem expects implementation of `pm_policy_state_lock_put/pm_policy_state_lock_get`.
@ceolin how do you feel with this proposal ?

Reply:
I found a better way to handle suspend action depending on driver state and if `CONFIG_PM_DEVICE_RUNTIME` is enabled. I'll push this soon